problem4.cpp

```diff
@@ -0,0 +1,22 @@
+#include <cstring>
+#include <iostream>
+
+int main() {
+  constexpr int Max_Char = 100;
+  std::cout << "Please, enter string (no more " << Max_Char
+            << " characters) \n";
+
+  char str[Max_Char]{};
+  std::cin >> str;
+
+  std::cout << "to upper: ";
+
+  for (size_t i{}; i < std::strlen(str); ++i) {
```

Comment:
ну если быть точным то вам и не нужен strlen. Что есть признаком конца строки? '\0' вот в цикле и делает проверку
```suggestion
  for (size_t i{}; str[i] != '\0'; ++i) {
```